src/mmap_unix.rs

```diff
@@ -20,15 +20,58 @@
 //! - [GuestMemoryMmap](struct.GuestMemoryMmap.html): provides methods to access a collection of
 //! GuestRegionMmap objects.
 
-use libc;
+use std::fs::File;
 use std::io;
 use std::ptr::null_mut;
 
+use libc;
+
+use guest_memory::FileMappingConfig;
 use mmap::AsSlice;
 use volatile_memory::{self, compute_offset, VolatileMemory, VolatileSlice};
 
 use std::os::unix::io::AsRawFd;
 
+// TODO: when mapping an file, should we also check that offset + length is smaller than the file
+// size? AFAIK, accesses past the end of the object can generate SIGBUS signals, but then again,
+// even if we do verify, someone might truncate the file later. I guess it might still be a
+// valuable sanity check.
+// TODO: should we explicitly check that the offset is a multiple of PAGE_SIZE, or will mmap return
+// an error anyway in this case?
```

Comment:
mmap will throw you an error, no need to check for this.

Reply:
Ah right, didn't remember the documentation actually mentions this.